x-pack/plugins/observability_solution/synthetics/server/common/inline_to_zip.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import archiver from 'archiver';
+import { MemWritable } from './mem_writable';
+
+function wrapInlineInProject(inlineJourney: string) {
+  return `import { journey, step, expect } from '@elastic/synthetics';
+
+journey('inline', ({ page, context, browser, params, request }) => {
+${inlineJourney}
+});`;
+}
+
+export async function inlineToProjectZip(
+  inlineJourney: string,
+  monitorId: string,
+  logger: Logger
+): Promise<string> {
+  const mWriter = new MemWritable();
+  try {
+    await new Promise<void>((resolve, reject) => {
+      const archive = archiver('zip', {
+        zlib: { level: 9 },
+      });
+      archive.on('error', reject);
+      mWriter.on('close', resolve);
+      archive.pipe(mWriter);
+      archive.append(wrapInlineInProject(inlineJourney), {
```

Comment:
I am not sure if this is a concern for Saved Objects, The zip source changes everytime when call this function mainly because of the Date metadata added by the archive library - We would need a dummy date to make that a no-op - https://github.com/elastic/synthetics/blob/10be1719ca5d37a39406ec2a2ed5f04d352720a5/src/push/bundler.ts#L76

Reply:
I'm fine with adding this.

I will say the procedure is likely not as important here as the only time this code is meant to run is at persist time (i.e. `add` or `update`). There aren't any cases where reads or other routes would cause the zip code to run, and we are already re-persisting the inline script anytime that happens. But it makes sense to me that we'd do this regardless.

See: 823143de82b76de4194f8d6d1f8b48b1b6c35d75